mlflow/models/display_utils.py

```diff
@@ -0,0 +1,115 @@
+import json
+
+import jinja2
+
+from mlflow.models.model import ModelInfo
+from mlflow.models.signature import ModelSignature
+from mlflow.utils import databricks_utils
+
+
+def _is_input_agent_compatible(inputs: list[dict]) -> bool:
+    messages = next(filter(lambda col: col.get("name") == "messages", inputs), None)
+    if not messages:
+        return False
+    if not messages.get("type") == "array":
+        return False
+    items = messages.get("items")
+    if not items:
+        return False
+    if items.get("type") != "object":
+        return False
+    properties = items.get("properties")
+    if not properties:
+        return False
+    if properties.get("content", {}).get("type") != "string":
+        return False
+    if properties.get("role", {}).get("type") != "string":
+        return False
+    return True
+
+
+def _is_output_string_response(outputs: list[dict]) -> bool:
+    content = next(filter(lambda col: col.get("name") == "content", outputs), None)
+    if not content:
+        return False
+    if content.get("type") != "string":
+        return False
+    return True
+
+
+def _is_output_string(outputs: list[dict]) -> bool:
+    return len(outputs) == 1 and outputs[0].get("type") == "string"
+
+
+def _is_output_chat_completion_response(outputs: list[dict]) -> bool:
```

Comment:
i wonder if there's an easier way to check if the input/output signatures are compatible. we have `CHAT_MODEL_INPUT_SCHEMA` and `CHAT_MODEL_OUPUT_SCHEMA` that you can probably compare against (not sure if a simple `==` works, but if it does then this code can be much simpler):

https://github.com/mlflow/mlflow/blob/4aead32f804798cdf7a8047ea26c138f85fca452/mlflow/types/llm.py#L737

however if you want to support a wider range of types by only checking a few keys, then i guess the current approach is the only way to go

Reply:
+1 this piece of code is hard to maintain.
What's our target customers? Should they use ChatModel? Or will this piece of code always work as long as the model signature is OAI compatible?
If we don't want to use ChatModel's signature, can we operate on the schema object directly instead of relying on the dictionary representation of the schema?

Reply:
We want this to shows up whenever the model supports the [agent evaluation schema](https://docs.databricks.com/en/generative-ai/agent-evaluation/evaluation-schema.html). The eval schema supports older agents, like langchain models with vanilla string, older `StringResponse`/`ChatRequest`, which is most of our customers. The agent eval schema is evolving very slowly, so keeping this in sync should be ok.

I'll change the code to operate on the schema object itself.

Reply:
Rewrote to use the schema object instead of a json dict